### PR TITLE
Http Parser

### LINKED ISSUE #5: Http Parser
Write a small Http Parser that would be able to extract the HTTP verb, path, query params and body.
Follow [this](https://www.jmarshall.com/easy/http/) guide.

The parser should:
- be instantiated with the raw HTTP string request: `HttpParser parser(myRawHttpRequest)`
- have members for saving the : Request type (Get/post/put/delete), path (/my/res/123), http version, query params (/my/res/path?value=123&value2=abs) in an unordered_map
- have member functions to extract all the data ^^^

### PR DESCRIPTION
**Pull request by stevenstonie**
🐱‍👤. 
for more info check the penultimate commit message.

### CODE REVIEW

File: Http/HttpParser.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#include "HttpParser.hpp"
+
+#include <Utils/Utils.hpp>
+
+HttpParser::HttpParser(const std::string& request) {
+}
+
+void HttpParser::extractRequestLineFromRequest(const std::string& request) {
```
Comment (Alex-dev02) [inline Http/HttpParser.cpp]: Numele functiilor ar trebui sa ifie PascalCase

File: Http/HttpParser.hpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+public:
+	HttpParser(const std::string& request);
+
+private:
+	std::string m_requestType;
+	std::string m_path;
+	std::string m_version;
+    std::unordered_map<std::string, std::string> m_queryParams;
+
+	void extractRequestLineFromRequest(const std::string& request);
```
Comment (Alex-dev02) [inline Http/HttpParser.hpp]: PascalCase

File: Utils/Utils.hpp [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+class Utils {
+   public:
+    static std::vector<std::string> split(const std::string& string, char delimiter);
+    static std::unordered_map<std::string, std::string> separateVectorOfKeysAndValues(const std::vector<std::string>& keyAndValues, char delimiter);
+};
```
Comment (Alex-dev02) [inline Utils/Utils.hpp]: Adauga un new line la sfarsitul oricarui fisier

File: Utils/Utils.cpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#include "Utils.hpp"
+
+std::vector<std::string> Utils::split(const std::string& text, char delimiter) {
```
Comment (Alex-dev02) [inline Utils/Utils.cpp]: PascalCase

File: Utils/Utils.cpp [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        } else {
+            token += elem;
+        }
+    }
+    tokens.emplace_back(token);
+    
+    return tokens;
+}
+
+std::unordered_map<std::string, std::string> Utils::separateVectorOfKeysAndValues(const std::vector<std::string>& keyAndValues, char delimiter) {
```
Comment (Alex-dev02) [inline Utils/Utils.cpp]: PascalCase

File: Utils/Utils.hpp [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+class Utils {
+   public:
+    static std::vector<std::string> split(const std::string& string, char delimiter);
+    static std::unordered_map<std::string, std::string> separateVectorOfKeysAndValues(const std::vector<std::string>& keyAndValues, char delimiter);
```
Comment (Alex-dev02) [inline Utils/Utils.hpp]: PascalCase

**Review by Alex-dev02 — CHANGES_REQUESTED** [review]

File: Http/HttpParser.cpp (hunk 1 above)
Comment (stevenstonie) [reply Http/HttpParser.cpp]: 🤢🤢

**Pull request merged**